packages/core/src/interfaces/truncation.ts

```diff
@@ -0,0 +1,19 @@
+export interface TruncationOptions {
+	/**
+	 * truncation configuration
+	 */
+	truncation?: {
+		/**
+		 * truncation type
+		 */
+		type?: string;
+		/**
+		 * truncation threshold
+		 */
+		threshold?: Number;
```

Comment:
what's the difference between `threshold` and `numCharacter`?

Reply:
Threshold here means the point where truncation applies. For example, if a word is 15 characters long and threshold is 10, it will be truncated. If a word is 5 characters long it will not be truncated. And for that truncated word I can set how many characters to be shown (`numCharacter`). We can just keep the `numCharacter` if you think `threshold` is unnecessary.